DYDX-Bots/Perpetual-Large-Deposits-Withdrawals/src/agent.spec.ts

```diff
@@ -0,0 +1,395 @@
+import { BigNumber } from "ethers";
+import { Interface } from "ethers/lib/utils";
+import { FindingType, FindingSeverity, Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { createAddress, MockEthersProvider, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { provideHandleTransaction } from "./agent";
+import BalanceFetcher from "./balance.fetcher";
+import { BotConfig } from "./config";
+import { ERC20_TOKEN_ABI, EVENTS } from "./utils";
+
+const createFinding = (name: string, args: any[], token: string) => {
+  if (name === "LogDeposit")
+    return Finding.fromObject({
+      name: "Large deposit into perpetual contract",
+      description: "LogDeposit event detected with large quantized Amount",
+      alertId: "DYDX-1-1",
+      protocol: "dYdX",
+      severity: FindingSeverity.Info,
+      type: FindingType.Info,
+      metadata: {
+        quantizedAmount: args[5].toString(),
+        starkKey: args[1].toHexString(),
+        token: token,
+      },
+    });
+  else if (name === "LogWithdrawalPerformed")
+    return Finding.fromObject({
+      name: "Large withdrawal into perpetual contract",
+      description: "LogWithdrawalPerformed event detected with large quantized Amount",
+      alertId: "DYDX-1-2",
+      protocol: "dYdX",
+      severity: FindingSeverity.Info,
+      type: FindingType.Info,
+      metadata: {
+        quantizedAmount: args[3].toString(),
+        token: token,
+        recipient: args[4].toLowerCase(),
+        ownerKey: args[0].toHexString(),
+      },
+    });
+  else
+    return Finding.fromObject({
+      name: "Large mint withdrawal into perpetual contract",
+      description: "LogMintWithdrawalPerformed event detected with large quantized Amount",
+      alertId: "DYDX-1-3",
+      protocol: "dYdX",
+      severity: FindingSeverity.Info,
+      type: FindingType.Info,
+      metadata: {
+        quantizedAmount: args[3].toString(),
+        token: token,
+        assetId: args[4].toString(),
+        ownerKey: args[0].toHexString(),
+      },
+    });
+};
+const createSuspiciousFinding = (name: string, token: string, args: any[]): Finding => {
+  let metadata =
+    name === "LogDeposit"
+      ? {
+          quantizedAmount: args[5].toString(),
+          starkKey: args[1].toHexString(),
+          assetType: token,
+        }
+      : { quantizedAmount: args[3].toString(), starkKey: args[0].toHexString(), assetType: token };
```

Comment:
- `assetType` is the `assetType` argument not the `token` in the suspicious finding case, I changed the parameter name to avoid this confusion. 

- Question answered in the previous conversation. I regrouped both under the same field because they refer to `stark key involved in the transfer` and that's the important info to give about the suspicious finding. The user can easily know if this key was the depositor / receiver / withdrawer from the event name.